### PR TITLE
Send only to internal email address in dev/stage

### PR DESCRIPTION
**Pull request by sayanarijit**

### CODE REVIEW

File: apphelpers/utilities/email.py [hunk 1]
```diff
@@ -90,6 +92,24 @@ def send_email(
...
+    if settings.DEBUG or settings.APP_MODE != "prod":
+        # Make sure that we don't send emails to non-@DOMAIN emails in dev/stage
+        filtered_recipients = []
+        for recpt in recipients:
+            if isinstance(recpt, (list, tuple)):
+                _, email = recpt
+            else:
+                email = recpt
+
+            if email.endswith(f"@{settings.DOMAIN}"):
```
Comment (sayanarijit) [inline apphelpers/utilities/email.py:104]: How to in allow .me in .net/.dev?

File: apphelpers/utilities/email.py [hunk 2]
```diff
@@ -90,6 +92,24 @@ def send_email(
...
+                filtered_recipients.append(recpt)
+            else:
+                app_logger.info(
+                    f"Skipping email to {email} as it does not end with @{settings.DOMAIN}"
+                )
+
+        recipients = filtered_recipients
```
Comment (shon) [inline apphelpers/utilities/email.py:115]: Other (slightly unreadable) way to write same code 

```Python
is_internal_email =  lambda recpt: (isinstance(recpt, str) and recpt or recpt[1]).endswith(settings.domain)
filtered_recipients = (recpt for recpt in recipients if is_internal_email(recpt))
```

File: apphelpers/utilities/email.py [hunk 3]
```diff
@@ -90,6 +92,29 @@ def send_email(
     """
     assert any((text, html)), "please provide html or text"
 
+    if settings.DEBUG or settings.APP_MPDE != "prod":
```
Comment (rajatsingla) [inline apphelpers/utilities/email.py:95]: Suggestion: we should also filter bcc list, 
another suggestion: Maybe this code is better suited at app level, as it is specific to our use case.

Comment (sayanarijit) [reply apphelpers/utilities/email.py:95]: bcc +1
The use case argument applies to the whole repo though.

**Pull request merged**